evalml/pipelines/components/transformers/preprocessing/time_series.py

```diff
@@ -56,16 +62,31 @@ def transform(self, X, y=None):
         Returns:
             pd.DataFrame: Transformed X.
         """
+        if X is None:
+            X = pd.DataFrame()
         # Normalize the data into pandas objects
-        if not isinstance(X, pd.DataFrame):
-            X = pd.DataFrame(X)
-        if y is not None and not isinstance(y, pd.Series):
-            y = pd.Series(y)
-
-        if self.delay_features and not X.empty:
-            X = X.assign(**{f"{col}_delay_{t}": X[col].shift(t)
-                            for t in range(1, self.max_delay + 1)
-                            for col in X})
+        X = _convert_to_woodwork_structure(X)
+        if y is not None:
+            y = _convert_to_woodwork_structure(y)
+
+        if y.logical_type == logical_types.Categorical:
+            original_y = _convert_woodwork_types_wrapper(y.to_series())
+            y_encoded = LabelEncoder().fit_transform(original_y)
+            y = pd.Series(y_encoded, index=original_y.index)
+        else:
+            y = _convert_woodwork_types_wrapper(y.to_series())
+
+        categorical_columns = {name for name, column in X.columns.items() if
+                               column.logical_type == logical_types.Categorical}
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+
+        if self.delay_features and len(X) > 0:
+            for col_name in X:
+                col = X[col_name]
+                if col_name in categorical_columns:
+                    col = LabelEncoder().fit_transform(col)
+                    col = pd.Series(col, index=X.index)
```

Comment:
Have you considered doing all the encoding over a subset of the data frame:
```
X[categorical_columns] = X[categorical_columns].apply(LabelEncoder().fit_transform)
```
Or something like that...I didn't run it ;)

Reply:
I'll give this a shot! Great suggestion :)